core/templates/domain/statistics/learner-action.model.ts

```diff
@@ -148,29 +114,37 @@ export class LearnerActionModel {
    * @returns {LearnerAction}
    */
   static createFromBackendDict(
-      learnerActionBackendDict: LearnerActionBackendDict): LearnerAction {
+      learnerActionBackendDict: LearnerActionBackendDict
+  ): LearnerAction {
     switch (learnerActionBackendDict.action_type) {
       case LearnerActionType.ExplorationStart:
-        return new ExplorationStartLearnerAction(
+        return new LearnerAction(
           learnerActionBackendDict.action_type,
-          learnerActionBackendDict.action_customization_args,
-          learnerActionBackendDict.schema_version);
+          learnerActionBackendDict
+            .action_customization_args as ExplorationStartCustomizationArgs,
+          learnerActionBackendDict.schema_version
+        );
       case LearnerActionType.AnswerSubmit:
-        return new AnswerSubmitLearnerAction(
+        return new LearnerAction(
           learnerActionBackendDict.action_type,
-          learnerActionBackendDict.action_customization_args,
-          learnerActionBackendDict.schema_version);
+          learnerActionBackendDict
+            .action_customization_args as AnswerSubmitCustomizationArgs,
+          learnerActionBackendDict.schema_version
+        );
       case LearnerActionType.ExplorationQuit:
-        return new ExplorationQuitLearnerAction(
+        return new LearnerAction(
           learnerActionBackendDict.action_type,
-          learnerActionBackendDict.action_customization_args,
-          learnerActionBackendDict.schema_version);
+          learnerActionBackendDict
+            .action_customization_args as ExplorationQuitCustomizationArgs,
+          learnerActionBackendDict.schema_version
+        );
       default:
         break;
     }
-    const invalidBackendDict: never = learnerActionBackendDict;
```

Comment:
Should the "never" be dropped here? (I'm not sure.)

Reply:
Removed